optuna/pruners/successive_halving.py

```diff
@@ -103,9 +119,17 @@ def prune(self, study, trial):
 
         rung = _get_current_rung(trial)
         value = trial.intermediate_values[step]
-        trials = None
+        trials = None  # type: Optional[List[FrozenTrial]]
 
         while True:
+            if self._min_resource is None:
+                if trials is None:
+                    trials = study.get_trials(deepcopy=False)
+                self._estimate_min_resource(trials)
+                if self._min_resource is None:
+                    return False
```

Comment:
Should we keep returning `True` when the value is NaN, even for the first trial (the NaN check comes after this return)?

Reply:
If return `True` here, the very first trial will be pruned. Is it acceptable?

Reply:
Sorry, not sure I'm understanding you correctly but my point was to not return here at all but to delay this check/return to after the NaN check to keep the existing behavior.

Reply:
> the NaN check comes after this return

The check exists there for passing mypy check.

Reply:
> Should we keep returning `True` when the value is NaN, even for the first trial (the NaN check comes after this return)?

This sounded like L.130 should return `True` instead of `False` to me...

Reply:
> Should we keep returning `True` when the value is NaN, even for the first trial (the NaN check comes after this return)?

This sounded like L.130 should return `True` instead of `False` to me...

Reply:
Sorry, it's probably correct the way it is, returning `False` here. (The rung promotion step below cannot be computed unless the minimum resource is determined.)

Reply:
> Sorry, not sure I'm understanding you correctly but my point was to not return here at all but to delay this check/return to after the NaN check to keep the existing behavior.

```python
def prune(self, study: Study, trial: FrozenTrial) -> bool:

    ...

    while True:
        if self._min_resource is None:
            if trials is None: trials = study.get_trials(False)
            self._min_resource = _estimate_min_resource(trials)

        if self._min_resource is None:
            return False
        else:
            ...
```

What you assume is like the above snippet?
I want to see what you think better in a snippet as I don't still get the point.

Reply:
Not really, that seems identical to the current implementation. As I wrote earlier,  https://github.com/optuna/optuna/pull/812/files#r364126282 however, you can ignore this comment as the current implementation looks correct.